### PR TITLE
fix-for-issue-#7665

### PR DESCRIPTION
**Pull request by MrHakimov**
Issue: #7665  - Add example for EmptyStatement

### CODE REVIEW

**Comment by rnveach** [comment]
@MrHakimov Please see https://github.com/checkstyle/checkstyle/wiki/PR-rules .
Specifically items 2, 3, and 4.

**Comment by MrHakimov** [comment]
> @MrHakimov Please see https://github.com/checkstyle/checkstyle/wiki/PR-rules .
> Specifically items 2, 3, and 4.

I did `mvn clean compile` and it showed `Build success`, I lose with squashing commits, because I didn't work with them, and I also don't know what to do with CI.

**Comment by rnveach** [comment]
> I did mvn clean compile

Please look at maven command again in point 3, it is `verify` not `compile`. Compile will just compile the code and will not run the tests.

> I also don't know what to do with CI

Anything with a red X is a failure. You can click on the details link next to each one to look at the logs and see where the failure is. It is basically a console output of running a maven command or similar.

>  I lose with squashing commits, because I didn't work with them

I am not following what you mean.Could you please clarify?

**Comment by MrHakimov** [comment]
> Please look at maven command again in point 3, it is `verify` not `compile`. Compile will just compile the code and will not run the tests.
> Oh, I see, will do that.
> Anything with a red X is a failure. You can click on the details link next to each one to look at the logs and see where the failure is. It is basically a console output of running a maven command or similar.
> Ok, thanks, will check it.
> I am not following what you mean.Could you please clarify?
> I didn't manage to understand what squashing is firstly, now I think I know hot to correct my commits.

**Comment by MrHakimov** [comment]
@rnveach, @strkkk, can you close this pull request, please?
I made a new branch and started from zero and now want to open new PR.

**Comment by MrHakimov** [comment]
> @rnveach, @strkkk, can you close this pull request, please?
> I made a new branch and started from zero and now want to open new PR.

Or can I close?

**Comment by MrHakimov** [comment]
@rnveach @strkkk
Can you check this issue now. It passed all of the tests now.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java [hunk 1]
```diff
@@ -40,6 +40,18 @@
...
+ * <div class="wrapper">
+ * <pre>
+ * class Expression {
+ *     List&lt;Expression&gt; arguments;
+ *     public void calculate() {
+ *         // Calculates something
+ *     }
+ * }
+ * for (int i = 0; i &lt; arguments.size(); arguments.get(i++).calculate()); // OK
+ * </pre>
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java]: there is no example of violations

**Comment by strkkk** [comment]
@MrHakimov please make sure your branch contains only 1commit. It looks like you rebased it incorrectly.

**Comment by MrHakimov** [comment]
@strkkk It seems that when I am pushing it adds one more commit, which is not mine.
It is `doc: fix indentation of config example`.

**Comment by strkkk** [comment]
@MrHakimov commit looks good now, please see rrview comment

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java [hunk 2]
```diff
@@ -40,6 +40,24 @@
...
+ *     public void calculate() {
+ *         // Calculates something
+ *     }
+ *
+ *     public bool isConstant() {
+ *         // Returns true, if Expression is constant, else returns false
+ *     }
+ * }
+ * for (int i = 0; i &lt; arguments.size(); arguments.get(i++).calculate()); // OK
+ * if (arguments.get(0).isConstant()); // Can violate NullPointerException if arguments or arguments[0] is null,
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java]: No need for such long description, `// violation` is enough. Please add more examples

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java [hunk 3]
```diff
@@ -40,6 +40,27 @@
  * <pre>
  * &lt;module name=&quot;EmptyStatement&quot;/&gt;
  * </pre>
+<<<<<<< HEAD
+=======
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java]: A merge artefact here. Please remove.

File: src/xdocs/config_coding.xml [hunk 4]
```diff
@@ -841,6 +841,27 @@ doUnconditionalStuff();
         <source>
 &lt;module name=&quot;EmptyStatement&quot;/&gt;
         </source>
+<<<<<<< HEAD
+=======
```
Comment (pbludov) [inline src/xdocs/config_coding.xml]: Same here

File: src/xdocs/config_coding.xml [hunk 5]
```diff
@@ -841,6 +841,27 @@ doUnconditionalStuff();
...
+    public bool isConstant() {
+        // Returns true, if Expression is constant, else returns false
+    }
+}
+for (int i = 0; i &lt; arguments.size(); arguments.get(i++).calculate()); // OK
+if (arguments.get(0).isConstant()); // Can violate NullPointerException if arguments or arguments[0] is null,
+                                    // or can violate IndexOutOfBoundsException, if arguments is empty.
+          </pre>
+        </div>
+>>>>>>> 5a1f024ab... Issue #7665: Update doc for EmptyStatement
```
Comment (pbludov) [inline src/xdocs/config_coding.xml]: and here

Comment (MrHakimov) [reply src/xdocs/config_coding.xml]: Done. Thanks)

**Comment by romani** [comment]
Please remove from your PR not your commit.

**Comment by MrHakimov** [comment]
> Please remove from your PR not your commit.

Done.

**Comment by MrHakimov** [comment]
> Please remove from your PR not your commit.

Oh, it doesn't work, I am trying to drop commit, but it fails.

**Comment by MrHakimov** [comment]
@rnveach @strkkk Can you check this PR again, please?

**Comment by strkkk** [comment]
@MrHakimov there are 3 commits in your PR, but should be only one.
You can - undo your commit, stash changes, drop 2 invalid commits, apply stashed changes and commit it, after it rebase on latest master.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java [hunk 6]
```diff
@@ -34,6 +34,24 @@
...
+ *         // Calculates something
+ *     }
+ *
+ *     public bool isConstant() {
+ *         // Returns true, if Expression is constant, else returns false
+ *     }
+ * }
+ * for (int i = 0; i &lt; arguments.size(); arguments.get(i++).calculate()); // OK
+ * while (true); // violation
+ * if (arguments.get(0).isConstant()); // violation
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java]: `else;` would be nice.

Also, code example is not compiliable anyway, you can remove class above.
examples can use some standard things like 
`arguments.get(0).isConstant()` -> `Math.random() > 0.5`
or
`arguments.size(); arguments.get(i++).calculate()` -> `i < 10; i++`

Comment (MrHakimov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java]: > `else;` would be nice.
> 
You mean that I can leave `else; // OK` as an example?
> Also, code example is not compiliable anyway, you can remove class above.
> examples can use some standard things like
> `arguments.get(0).isConstant()` -> `Math.random() > 0.5`
I do not see logic behind `if (Math.random() > 0.5);`, for example.
> or
> `arguments.size(); arguments.get(i++).calculate()` -> `i < 10; i++`

**Comment by romani** [comment]
@MrHakimov, please remove from your branch not your commits

**Comment by rnveach** [comment]
@MrHakimov Please look at https://github.com/checkstyle/checkstyle/pull/7674/commits for the commits in your PR that we see. When that list shows only 1 commit, we can continue.

**Comment by romani** [comment]
@MrHakimov, please cache your change in some place. Drop your branch , pull latest master from main repo. Create new branch on your local, apply your changes, push to remote with the same name of branch with force option.

**Comment by romani** [comment]
There will be no review till you make CI green, and format your PR description as requested in issue.

**Comment by MrHakimov** [comment]
I don't have idea how to do that for now.
Also it seems that I lose development branch at checkstyle
image.
I don't know what to do, please, help :)
And yes, it passes all of the tests in my local directory.
image

**Comment by rnveach** [comment]
> And yes, it passes all of the tests in my local directory.

Are you running `mvn clean verify` ?

> Also it seems that I lose development branch at checkstyle

I would read some documents/videos on git and github.

I've never seen this and it looks like somehow github lost connection with your fork. Maybe you deleted it? Your branch at https://github.com/MrHakimov/checkstyle/commits/development looks correct but it is not the version in this PR. We need your latest version in a PR.

**Comment by MrHakimov** [comment]
> Are you running `mvn clean verify` ?

Yes.
> I would read some documents/videos on git and github.

I have already looked for a lot of information on that.
> I've never seen this and it looks like somehow github lost connection with your fork. Maybe you deleted it? Your branch at https://github.com/MrHakimov/checkstyle/commits/development looks correct but it is not the version in this PR. We need your latest version in a PR.

Ohhhhhh, yes. I suppose, I have deleted my repo and then forked again.

**Comment by MrHakimov** [comment]
What can you advise me to do?

**Comment by rnveach** [comment]
@MrHakimov Since it looks like github lost track of your branch, I would close this and create another PR. I don't know how else to fix it. Do not delete your fork or branch anymore until the PR is reviewed and merged.

**Pull request closed without merging**